tests/dualstack_test.py

```diff
@@ -128,6 +134,22 @@ def test_dualstack(self) -> None:
                 self.server_xds_host, self.server_xds_port
             )
 
+        with self.subTest("_start_firewall_rule_creation_thread"):
+
+            def recreate_firewall_rule():
+                time.sleep(30)
+                while self.firewall_rule_creation_should_stop is False:
```

Comment:
Please use [`threading.Event`](https://docs.python.org/3/library/threading.html) for inter-thread synchronization. The current approach may suffer from tearing.